node/impl/eth/transaction.go

```diff
@@ -397,6 +397,9 @@ func (e *ethTransaction) EthGetTransactionReceiptLimited(ctx context.Context, tx
 	} else if msgLookup == nil {
 		// This is the best we can do. In theory, we could have just not indexed this
 		// transaction, but there's no way to check that here.
+		if limit > api.LookbackNoLimit {
+			return nil, xerrors.Errorf("transaction receipt not found or may be too old (limit: %d epochs)", limit)
+		}
```

Comment:
```suggestion
		// This is the best we can do. We may just not have indexed this transaction, or we may have a
		// limit applied and not searched far back enough, but we don't have a way to go. Because
		// Ethereum tooling expects an empty response for transaction-not-found, we don't have a way of
		// differentiating between "can't find" and "doesn't exist".
```

Sorry @ameeetgaikwad, even though you did the hard work here https://github.com/filecoin-project/lotus/pull/12999 came up in the meantime and clarified that we need to return `nil` because it may be that it doesn't exist, even with the limit. So we'll put this comment in here for now and perhaps revisit it if we run into troubles with people not being able to find old transactions.